### PR TITLE
SALTO-3630: Partial Fetch Bugfix

### PR DESCRIPTION
**Pull request by tamtamirr**
As part of testing out Partial Fetch with my recent `Fetch With Change Detection` locally, I've noticed that a partial fetch on a certain type, will remove the `generatedDependencies` to `InstalledPackage` instances if they were not part of the `fetch targets`.

You can see the Workspace diff when attempting to partially fetch `ApexClass`: https://github.com/salto-io/tamir-sf/commit/f864ea77c6cb9ec3624a93f47d4cdecc2d7974b8

---

_Release Notes_: 
_None_

---
_User Notifications_: 
_None_

### CODE REVIEW

File: packages/salesforce-adapter/src/fetch_profile/metadata_query.ts [hunk 1]
```diff
@@ -97,12 +110,11 @@ export const buildMetadataQuery = (
       ? isFolderMetadataTypeNameMatch(instance, name)
       : regex.isFullRegexMatch(instance.name, name)
   }
-
   const isIncludedInPartialFetch = (type: string): boolean => {
     if (target === undefined) {
       return true
     }
-    if (target.includes(type)) {
+    if (target.includes(type) || ALWAYS_INCLUDED_IN_PARTIAL_FETCH.includes(type)) {
```
Comment (tamtamirr) [inline packages/salesforce-adapter/src/fetch_profile/metadata_query.ts:117]: Since **MANY** MetadataTypes can possibly come from InstalledPackage, it's better to always retrieve them in partial fetch.

File: packages/salesforce-adapter/src/filters/create_missing_installed_packages_instances.ts [hunk 2]
```diff
@@ -37,6 +37,9 @@ const createMissingInstalledPackageInstance = (file: FileProperties, installedPa
 const filterCreator: RemoteFilterCreator = ({ client, config }) => ({
   name: 'createMissingInstalledPackagesInstancesFilter',
   onFetch: async (elements: Element[]): Promise<FilterResult | undefined> => {
+    if (config.fetchProfile.metadataQuery.isPartialFetch()) {
+      return
+    }
```
Comment (ori-moisis) [inline packages/salesforce-adapter/src/filters/create_missing_installed_packages_instances.ts:42]: why was this added?
what if the fetch target includes installed packages?

File: packages/salesforce-adapter/src/filters/create_missing_installed_packages_instances.ts [hunk 3]
```diff
@@ -37,6 +37,9 @@ const createMissingInstalledPackageInstance = (file: FileProperties, installedPa
 const filterCreator: RemoteFilterCreator = ({ client, config }) => ({
   name: 'createMissingInstalledPackagesInstancesFilter',
   onFetch: async (elements: Element[]): Promise<FilterResult | undefined> => {
+    if (config.fetchProfile.metadataQuery.isPartialFetch()) {
```
Comment (tamtamirr) [inline packages/salesforce-adapter/src/filters/create_missing_installed_packages_instances.ts:40]: This can lead to removal of the `version` of `InstalledPackage` instances we actually fetch.
This is also relevant for `fetchWithChangeDetection` (quick fetch) as a followup to other tests I ran around my DRAFT PR

File: packages/salesforce-adapter/src/filters/create_missing_installed_packages_instances.ts (hunk 2 above)
Comment (tamtamirr) [reply packages/salesforce-adapter/src/filters/create_missing_installed_packages_instances.ts:42]: Just commented on it asynchronously. This is also relevant for `fetchWithChangeDetection` flows. (Where I combined the meaning of `isPartialFetch` to be both for Targeted fetches and Quick fetches) but it can be handled in the other PR, I'll change that to check if `InstalledPackage` is included in the fetch instead

Comment (ori-moisis) [reply packages/salesforce-adapter/src/filters/create_missing_installed_packages_instances.ts]: but why is this needed at all?
the next lines will effectively do this (because in a partial fetch without installed packages, the installed packages type won't be in `elements`)
even if the change itself is ok, I don't see why it needs to be in this PR with the bugfix, it doesn't seem related

Comment (tamtamirr) [reply packages/salesforce-adapter/src/filters/create_missing_installed_packages_instances.ts]: will remove

**Review by ori-moisis — APPROVED** [review]
